pkg/webhooks/agentinstall/v1beta1/agent_service_config_admission_hook.go

```diff
@@ -0,0 +1,324 @@
+package v1beta1
+
+import (
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/openshift/assisted-service/api/v1beta1"
+	log "github.com/sirupsen/logrus"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	agentServiceConfigResource         = "agentserviceconfigs"
+	agentServiceConfigAdmissionGroup   = "admission.agentinstall.openshift.io"
+	agentServiceConfigAdmissionVersion = "v1"
+)
+
+// Minimum sizes of the storage areas:
+var (
+	minDatabaseStorage   = resource.MustParse("50M")
+	minFilesystemStorage = resource.MustParse("50M")
+	minImageStorage      = resource.MustParse("10G")
```

Comment:
These sizes are just examples, I will adjust them to the real values before merging this.

Reply:
I think we want these configurable. 10G image storage, for instance, may be too large for some systems